src/main/java/com/faforever/client/chat/ChatUserItemController.java

```diff
@@ -192,11 +223,43 @@ private void setAvatarUrl(@Nullable String avatarUrl) {
   }
 
   private void setClanTag(String newValue) {
+    // code to test can be inserted here set new Value to some existing clanTag and the players clan...player.setClan("TAG")
+   
     if (StringUtils.isEmpty(newValue)) {
-      clanLabel.setVisible(false);
+      clanMenu.setVisible(false);
     } else {
-      clanLabel.setText(String.format(CLAN_TAG_FORMAT, newValue));
-      clanLabel.setVisible(true);
+
+
+      Runnable setMenuItems = new Runnable() {
+        @Override
+        public void run() {
+          clan = clanService.getClanByTag(player.getClan());
+
+          if (clan != null) {
+            MenuItem page = new MenuItem(i18n.get("clan.visitPage"));
+            page.setOnAction(event -> {
+              platformService.showDocument(baseClanWebsite + clan.getClanId());
+              // TODO: Could be viewed in clan section (if implemented)
+            });
+
+            if (playerService.isOnline(clan.getLeaderName())) {
+              MenuItem toLeader = new MenuItem(i18n.get("clan.toLeader"));
+              toLeader.setOnAction(event -> {
+                onClanTagClicked();
+              });
```

Comment:
IntelliJ colorizes these `{` `}` and if you select one of them, press Alt+Enter it will recommend you to remove them because they are not needed - making the code a bit cleaner